File: salford/gmdspconverters/parking.rb
require 'rdf'
require 'rdf/rdfxml'
require 'open-uri'
require 'csv'
require './osgbconvert'

include RDF

COUNCIL = "salford"

# add the data to the graph
graph = RDF::Graph.new

# set up the vocabulary
RDFS = RDF::RDFS
GEO = RDF::GEO
VCARD = RDF::VCARD
OS = RDF::Vocabulary.new("http://data.ordnancesurvey.co.uk/ontology/admingeo/")

# our new vocabulary
PARKING = RDF::Vocabulary.new("http://data.gmdsp.org.uk/id/" + COUNCIL + "/parking/")

def idify(s)
  rs = s.downcase
  rs.gsub!(" ", "-")
  rs
end


CSV.foreach('CarParks.csv', { headers:true }) do |csv_obj|
  label = csv_obj["Name"]
  # create a unique name
  subject = PARKING[idify(label)]
  # add the label
  graph << [subject, RDFS.label, label]
  # add the recycling centre type
  graph << [subject, RDF.type, RDF::URI("http://data.gmdsp.org.uk/def/parking")]
  # add the address location
  graph << [subject, VCARD.hasStreetAddress, csv_obj["Address"]]
  # location information
  osgb36point = convertWGS84toOSGB36(csv_obj["Latitude"].to_f, csv_obj["Longitude"].to_f, 0)
  locations = LatLongToOSGrid(osgb36point[0], osgb36point[1])
  graph << [subject, OS.easting, locations[0]]
  graph << [subject, OS.northing, locations[1]]

  # parking type
  graph << [subject, PARKING.type, csv_obj["Type"]] unless csv_obj["Type"].nil?

  # Operator info - should this be a organisation type?
  graph << [subject, PARKING.operator, csv_obj["Operator"]] unless csv_obj["Operator"].nil?
  graph << [subject, PARKING.operatorUrl, csv_obj["URL"]] unless csv_obj["URL"].nil?


end


RDF::RDFXML::Writer.open("carparks.rdf") do |writer|
  writer << graph
end

File: salford/gmdspconverters/parking.py
__author__ = 'jond'

import csv

from rdflib import URIRef, Literal, Namespace, RDF

from gmdspconverters import utils

PARKING = Namespace('http://data.gmdsp.org.uk/id/salford/parking/')
PARKING_ONT = Namespace('http://data.gmdsp.org.uk/def/council/parking/')

def convert(graph, input_path):
    reader = csv.DictReader(open(input_path, mode='r'))

    for row in reader:
        p = PARKING[utils.idify(row["Name"])]
        graph.add((p, RDF.type, PARKING_ONT["ParkingSite"]))
        graph.add((p, utils.RDFS['label'], Literal(row["Name"])))
        graph.add((p, PARKING_ONT['type'], Literal(row["Type"])))
        graph.add((p, PARKING_ONT['informationPage'], Literal(row["URL"])))

        p_operator = PARKING["operator/" + utils.idify(row["Operator"])]
        graph.add((p, PARKING_ONT['operator'], p_operator))
        graph.add((p_operator, RDF.type, PARKING_ONT["ParkingSiteOperator"]))
        graph.add((p_operator, utils.RDFS['label'], Literal(row["Operator"])))
        graph.add((p_operator, utils.ADMINGEO["MetropolitanDistrict"], URIRef("http://data.ordnancesurvey.co.uk/id/7000000000018805")))

        graph.add((p, utils.GEO["lat"], Literal(row["Latitude"])))
        graph.add((p, utils.GEO["long"], Literal(row["Longitude"])))

        address = utils.idify(row["Address"])
        graph.add((p, utils.VCARD['adr'], URIRef("http://data.gmdsp.org.uk/def/council/parking/address/"+address)))

        # now add the address VCARD
        vcard = PARKING["address/"+address]
        graph.add((vcard, RDF.type, utils.VCARD["location"]))
        graph.add((vcard, utils.RDFS['label'], Literal(row["Name"])))
        graph.add((vcard, utils.VCARD['street-address'], Literal(row["Address"])))
        graph.add((vcard, utils.VCARD['postal-code'], Literal(row["Postcode"])))
